docker.go

```diff
@@ -138,7 +144,9 @@ func (p Plugin) Exec() error {
 		trace(cmd)
 
 		err := cmd.Run()
-		if err != nil {
+		if err != nil && cmd.Args[1] == "pull" {
```

Comment:
Can we place this code in a helper function (e.g. isPullCache) and check index size to prevent an index out of bound?

Reply:
this shouldn't go out of bounds as is, if you consider that Args[0] is docker and Args[1] has to be something at all times

Reply:
unchecked bounds should be avoided, as should magic number of this sort. Moving this to a named function also makes the code more readable. Otherwise this code may make no sense to anyone that hasn't read this thread.

Reply:
My initial thinking was the same as @shimunn but your points are definitely valid and ultimately better for the reasons you stated.  Looks like @shimunn issued a PR against my branch in my fork with these changes. Ill get that resolved there so this can move forward